### ISSUE TITLE
Dynamically set username and password

### ISSUE THREAD

**Issue by karlingen** [jage/elk#9]
Is it possible to dynamically set username and password with this gem without having to initialize the entire app with predefined values inside `config/initializers/elk.rb`?

For instance, inside a controller I would like to set the username and password to whatever value the user submits. 

Would it be possible to simply use this code inside the controller (or whatever class):

```ruby
def whatever_action
  require 'elk'
  Elk.configure do |config|
    config.username = params['some_value']
    config.password = params['some_other_value']
  end
  Elk::SMS.send({
      from: 'foobar company',
      to: '+46700000000',
      message: 'foobar',
      whendelivered: 'someurl.com/'
    })
end
```

**Comment by karlingen** [comment]
Seems to be working :)

**Comment by jage** [comment]
@karlingen Yeah that will work. but I would be careful with doing this, I assume this causes issues in a threaded environment (haven't looked tried this yet).

I'd like to improve this, so multiple instances can live simultaneously in a thread safe manner. I'll leave this issue open and come back to you when I've looked into this.

**Comment by karlingen** [comment]
Sounds great! :+1:

**Comment by jage** [comment]
Sorry for the delay @karlingen
I've just released a version with the changes in #13 (0.2.0).

With the introduction of `Elk::Client` you could solve this with something like:

```ruby
require 'elk'

def whatever_action
  client = Elk::Client.new
  client.configure do |config|
    config.username = params['some_value']
    config.password = params['some_other_value']
  end
  Elk::SMS.send({
      client: client,
      from: 'foobar company',
      to: '+46700000000',
      message: 'foobar',
      whendelivered: 'someurl.com/'
    })
end
```

This should be much safer and better than using the old configuration code.

**Comment by karlingen** [comment]
Thanks @jage  :heart:
